source/frontend/src/features/mapSideBar/disposition/tabs/offersAndSale/dispositionSale/update/UpdateDispostionSaleView.tsx

```diff
@@ -0,0 +1,121 @@
+import axios, { AxiosError } from 'axios';
+import { Formik } from 'formik';
+import styled from 'styled-components';
+
+import LoadingBackdrop from '@/components/common/LoadingBackdrop';
+import { DispositionSaleFormModel } from '@/features/mapSideBar/disposition/models/DispositionSaleFormModel';
+import SidebarFooter from '@/features/mapSideBar/shared/SidebarFooter';
+import { getCancelModalProps, useModalContext } from '@/hooks/useModalContext';
+import { IApiError } from '@/interfaces/IApiError';
+import { Api_DispositionFileSale } from '@/models/api/DispositionFile';
+
+import DispositionSaleForm from '../form/DispositionSaleForm';
+import { DispositionSaleFormYupSchema } from '../form/DispositionSaleFormYupSchema';
+
+export interface IUpdateDispositionSaleViewProps {
+  initialValues: DispositionSaleFormModel;
+  loading: boolean;
+  onSave: (sale: Api_DispositionFileSale) => Promise<Api_DispositionFileSale | undefined>;
+  onCancel: () => void;
+  onSuccess: () => void;
+  onError: (e: AxiosError<IApiError>) => void;
+}
+
+const UpdateDispositionSaleView: React.FC<IUpdateDispositionSaleViewProps> = ({
+  initialValues,
+  loading,
+  onSave,
+  onCancel,
+  onSuccess,
+  onError,
+}) => {
+  const { setModalContent, setDisplayModal } = useModalContext();
+
+  const cancelFunc = (resetForm: () => void, dirty: boolean) => {
+    if (!dirty) {
+      resetForm();
+      onCancel();
+    } else {
+      setModalContent({
+        ...getCancelModalProps(),
+        handleOk: () => {
+          resetForm();
+          setDisplayModal(false);
+          onCancel();
+        },
+      });
+      setDisplayModal(true);
+    }
+  };
+
+  return (
+    <StyledFormWrapper>
+      <Formik<DispositionSaleFormModel>
+        enableReinitialize
+        validationSchema={DispositionSaleFormYupSchema}
+        initialValues={initialValues}
+        onSubmit={async (values: DispositionSaleFormModel, formikHelpers) => {
+          try {
+            const sale = await onSave(values.toApi());
+            if (sale && sale.id) {
```

Comment:
note that the sale id could be zero

Reply:
that's ok, when Zero it would mean the Api request call wasn't successful.

Reply:
under that logic it should check for a zero as well

Reply:
updated